app/services/process_state.rb

```diff
@@ -0,0 +1,45 @@
+# ProcessState is an *experimental* thing that infers the state from
+# the state of application choices. Do not rely on this for business logic.
+class ProcessState
+  def initialize(application_form)
+    @application_form = application_form
+  end
+
+  def state
+    if states.empty?
+      :unsubmitted
+    elsif unique_states == %w[unsubmitted]
```

Comment:
We might be more explicit here, to check that all the states are unsubmitted we create use a method called `all_states_are?`

for example:
`all_states_are?(:unsubmitted)`
and in a similar way `all_states_are(:application_complete)`, and so on

Reply:
Good idea, done.